### PR TITLE
Change VPC CR Conditions

### PR DESCRIPTION
**Pull request by lxiaopei**
For IPAddressAllocation, SecurityPolicy, StaticRoute, SubnetPort CR, change to use "reason" for simplified message and "message" for details messages. according to https://github.com/kubernetes/apimachinery/blob/d4f471b82f0a17cda946aeba446770563f92114d/pkg/apis/meta/v1/types.go#L1407 Reason : reason contains a programmatic identifier indicating the reason for the condition Message : message is a human readable message indicating details about the transition

### CODE REVIEW

File: pkg/controllers/subnetport/subnetport_controller.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 		nsxSubnet, err := r.SubnetService.GetSubnetByPath(nsxSubnetPath)
 		if err != nil {
-			updateFail(r, ctx, subnetPort, &err)
+			c(r, ctx, subnetPort, &err)
```
Comment (yanjunz97) [inline pkg/controllers/subnetport/subnetport_controller.go:115]: accidental modification?

File: pkg/controllers/securitypolicy/securitypolicy_controller.go [hunk 2]
```diff
@@ -253,13 +253,13 @@ func (r *SecurityPolicyReconciler) setSecurityPolicyReadyStatusTrue(ctx context.
 func (r *SecurityPolicyReconciler) setSecurityPolicyReadyStatusFalse(ctx context.Context, secPolicy *v1alpha1.SecurityPolicy, transitionTime metav1.Time, err *error) {
 	newConditions := []v1alpha1.Condition{
 		{
-			Type:    v1alpha1.Ready,
-			Status:  v1.ConditionFalse,
-			Message: "NSX Security Policy could not be created/updated",
-			Reason: fmt.Sprintf(
+			Type:   v1alpha1.Ready,
```
Comment (timdengyun) [inline pkg/controllers/securitypolicy/securitypolicy_controller.go:256]: https://github.com/vmware-tanzu/nsx-operator/pull/791/files#diff-f45f721ba06ab477de36df3522bb24718a5f69d5747f3d6b5ae543fe98d68877L245
Seemly, as the new convention to update status reason/message, we should also update the StatusReady True condition as well

Comment (timdengyun) [reply pkg/controllers/securitypolicy/securitypolicy_controller.go:256]: This is also apply to other controllers.

File: pkg/controllers/subnetport/subnetport_controller.go (hunk 1 above)
Comment (lxiaopei) [reply pkg/controllers/subnetport/subnetport_controller.go:115]: yea,changed back

**Review by timdengyun — DISMISSED** [review]
LGTM

**Review by yanjunz97 — DISMISSED** [review]

**Comment by yanjunz97** [comment]
Looks like the UT regarding status update in ipaddressallocation fails

**Comment by lxiaopei** [comment]
fixed now. thanks.

**Review by yanjunz97 — APPROVED** [review]

**Review by timdengyun — APPROVED** [review]
LGTM

**Pull request merged**